### ISSUE TITLE
Can't click after the system was under heavy load

### ISSUE THREAD

**Issue by pclouds** [pekdon/pekwm#104]
This has happened on two machines of mine so it can't be hw specific. Somtimes I need to run some really heavy compilation that results in a lot of swapping and the desktop could become really laggy for a while. But that's not the problem. The problem is after that happens, I could type (and switch windows using keyboard) and move the mouse around, but I cannot click anything anymore. The only workaround is to switch to another workspace, then I can start clicking again.

Any idea what may cause it, or what sort of debug I should turn on to help catch it?

**Comment by biopsin** [comment]
Have you tried start with `pekwm --log-file /tmp/pekwm_issue.log --log-level trace` ?
However I have had some oddball input blockade very seldom, but I escape to tty* and back to unblock it.

**Comment by pclouds** [comment]
Thanks! I haven't been able to reproduce my problem yet, but at least the trace shows switching workspaces involves X11::ungrabServer and X11::grabServer. That does seem to suggest some problem with ungrabbing (or forgetting to do so)...

**Comment by biopsin** [comment]
> That does seem to suggest some problem with ungrabbing (or forgetting to do so)...
You're right about that, as I've being using mostly kb past months and no lockups even under heavy loads.
There's something with the pointer input where things block. But this bug goes unfortunetly beyond my understanding. Either the long way of bisect or gdb might be a last resort. What do you think pclouds?

**Comment by pclouds** [comment]
Problem is I can't reliably reproduce to even bisect... Haven't spent much time on this lately but I probably can look more into this grabbing business, then maybe see if i can add some trace code or something to help track it down

**Comment by pigge2** [comment]
if you do not change workspace does it work if you only hit the "alt" key? because that is a problem i have like once or a couple of times a day (although, it does not require heavy load) pekdon fixed the crashing (which sometimes were related to the lockup's in my case)

**Comment by pclouds** [comment]
Hmm.. can't remember if I tried alt key. But I will next time it happens and report back.

**Comment by pclouds** [comment]
Nope. Alt didn't help

**Comment by biopsin** [comment]
Ive also noticed another glitch lately that I cant launch menus with rightclick or expand toolbar items in any clients exept on the pekwm desktop, a pekwm restart resolves the problem.

**Comment by pekdon** [comment]
Closing this a duplicate of #4 , it seems to be related to how buttons are grabbed and then later replayed with XAllowEvents.
